src/app/Http/Controllers/API/AuthController.php

```diff
@@ -106,4 +112,57 @@ public function user(Request $request)
     {
         return new UserResource($request->user());
     }
+
+  /**
+     * Send an email to the user with the reset password link
+     *
+     * @bodyParam email email required email of the user who makes the request
+     *
+     * @param Request $request
+     */
+    public function sendResetLinkEmail(Request $request)
+    {
+        $request->validate([
+            'email' => 'required|email|exists:users,email'
+        ]);
+
+        // generate and save the token
+        $token = Str::random(60);
+        DB::table('password_resets')->insert([
+            'email' => $request->email,
+            'token' => $token,
+            'created_at' => Carbon::now()
+        ]);
+
+        Mail::to($request->email)->send(new ResetPasswordMail($token));
+    }
+
+    /**
+     * Change the password
+     *
+     * @bodyParam token string required token to validate the request
+     * @bodyParam email email required email of the user who wants to change is password
+     * @bodyParam password string required the new password for the user
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function reset(Request $request)
+    {
+        $request->validate([
+            'token' => 'required',
+            'password' => 'required|string|min:6',
+            'email' => 'required|email'
```

Comment:
Shouldn't you the same as `sendResetLinkEmail` and check here the existence of the email?
`'email' => 'required|email|exists:users,email'`

Reply:
yes, mb, fixed